### ISSUE TITLE
Add support for VSTS.TCM.AutomatedTestName

### ISSUE THREAD

**Issue by bryanbcook** [bryanbcook/azdevops-testplan-extension#20]
Automated Tests can be [associated to Test Cases using Visual Studio](https://learn.microsoft.com/en-us/azure/devops/test/associate-automated-test-with-test-case?view=azure-devops#associate-your-test). These values can also be assigned through the Azure DevOps REST API.

When associating tests, the following properties are set:

* Microsoft.VSTS.TCM.AutomatedTestName: The name or identifier of the automated test associated with the test case
* Microsoft.VSTS.TCM.AutomatedTestStorage: The assembly that contains the automated test case
* Microsoft.VSTS.TCM.AutomatedTestId: unique identifier
* Microsoft.VSTS.TCM.AutomatedTestType: descriptor for the test type, "Unit Test", "Coded UI", etc
* Microsoft.VSTS.TCM.AutomationStatus: Automated, Not Automated, Planned

The logic to generate the unique identifier is described in this log: https://devblogs.microsoft.com/devops/how-to-associate-automation-programmatically/

Expected behaviour:

* **.NET test automation using NUnit, xUnit, MSTest formats**:
  * developers should be able to use the associate to test case feature of Visual Studio
  * the _Microsoft.VSTS.TCM.AutomatedTestName_ in the TestCase should automatically line up to the `testcase.name` property from the test-result-parser library as this typically represents the full-class-name-method in those formats.
  *  if the test class, method or assembly are modified, users would need to re-associate the test case.

* **Other platforms (JUnit, Mocha, Cucumber)**:
  * Out of scope for now

**Comment by C0nquistadore** [comment]
I would gladly appreciate this, because we are currently using the VSTest@2 task to run our tests on windows. But now we are switching over some tests to linux/docker where the task is no longer supported, but we still want to be able to associate the test results with the test cases/test points.
As we already have the AutomatedTestName set on all test cases for the VSTest task to work, it would make the transition to this task a lot easier, if it would be automatically picked up.
Do you have a workaround for this until this is implemented? My first guess would be to use a regex because some of our test method names match the format "TC_<TestCaseId>_*". Or do you have a better suggestion?

**Comment by bryanbcook** [comment]
I'm glad to hear that you've found a valid use case for this. I just assumed that if teams were using the test automation property they wouldn't need this extension and would simply use VSTask. The VSTask has a windows executable that performs the look-up (and also has some awesome test sharding features for parallel execution). I hadn't considered the Windows to Linux conversion scenario.

I'd be happy to look at adding this, but I'll need some time to create a reproducible scenario. Also, I'm confident that  it'll work for TRX, because .NET has fully-qualified test case names, but other frameworks don't, so the solution would likely only use `Microsoft.VSTS.TCM.AutomatedTestName` and we'd need guidance for other platforms if they intend to use this. I should be able to find some cycles for this, hopefully before month's end. I'd also welcome pull requests.

> Do you have a workaround for this until this is implemented? 

Adding the TestCaseId to the name of the Test would be my first recommendation for a workaround.

> The problem is that we create a new test plan for each iteration which clones all the test cases and they end up with a new ID that doesn't match the associated test name anymore.

Unless you are deliberately making duplicate Test Cases and then manually re-associating the `Microsoft.VSTS.TCM.AutomatedTestName`, that's not my understanding. Test Cases and their IDs are fixed. When you create a new Test Plan and copy Test Suites into it, new Test Suites are created, but they include references to the original Test Cases. Within a Test Plan, the uniqueness of the TestPlan+TestConfig+TestCase is used to establish the TestPoint. To illustrate this, link a Test Case in a Test Suite to a User Story, then clone the Test Suite. The Story will only have one TestedBy association. You can get the TestCaseId in the URL when you edit the TestCase definition. If you're seeing something else, I'd be curious to learn how your team is setting this up.

**Comment by C0nquistadore** [comment]
Thanks for the quick reply!
> I should be able to find some cycles for this, hopefully before month's end.

No rush, I am currently fine with porting tests to linux and view the test results within the release pipeline using the `PublishTestResults@2` task. I am just glad that this is actually possible because Microsoft doesn't seem to have this scenario anywhere in their sight.
Having support for `Microsoft.VSTS.TCM.AutomatedTestName` and attachments (which we rely on) would be optimal for us to switch over to your extension.

> I'd also welcome pull requests.

I played around with the REST API already locally using C# and at least the automated test name feature doesn't look too complicated but Typescript ain't my daily work therefore it would also take me a while when I find the time.

> I'd be curious to learn how your team is setting this up.

Basically we clone the test plan for each iteration using these steps:

![Untitled3](https://github.com/user-attachments/assets/4e6c8bb9-4c35-484e-bee4-9968a8a4016c)

![Untitled2](https://github.com/user-attachments/assets/bfcca8ce-29f6-4368-8d5d-28148cd1431b)

![Untitled1](https://github.com/user-attachments/assets/52060167-3a74-4f9f-8ca1-b924423649e3)

In this scenario the test cases are cloned and therefore get new IDs while the automated test name stays the same. We then clone the release pipeline and/or adjust it the VSTest task to the new testplan and avoid having to update all method names in the source code.

I am not a big fan of the cloning to be honest and it wasn't my decision but it's not really my focus to look for an alternative.

**Comment by bryanbcook** [comment]
Shame about cloning the test cases, but I definitely see the value of using the automated test name for your scenario.

A few other bugs came in this weekend. Hopefully I can carve some time this weekend for these.

**Comment by bryanbcook** [comment]
@C0nquistadore This is now available! 

I was able to perform some limited testing on my end, but I would to hear feedback if this is working for you.

**Comment by C0nquistadore** [comment]
I have tried this out without any specific configuration (using `testCaseMatchStrategy: auto`) and it works great! Thanks!
